presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -139,6 +144,39 @@ property
     : identifier EQ expression
     ;
 
+parameterDeclaration
+    : identifier type
+    ;
+
+routineCharacteristics
+    : routineCharacteristic*
+    ;
+
+routineCharacteristic
+    : language
+    | determinism
+    | sqlDataAccess
+    | nullCallClause
+    ;
+
+language
+    : SQL
+    ;
+
+determinism
+    : DETERMINISTIC
+    | NOT DETERMINISTIC;
+
+sqlDataAccess
```

Comment:
I'm inclined to not include this for now because there's only one option and there's no use of this data. It's always safe to assume when language is "SQL" it will be "contains sql" anyways.

Reply:
`SqlDataAccess` can also be `READS SQL DATA` and `MODIFIES SQL DATA` when language is `SQL`. Since this is trivial to add, we should just do it.

Syntax rules only dictates how the query text is structured and is not a bad thing to provide the flexibility to use to include or exclude this clause.

It's also safe to assume that the language is `SQL` when the body is a return statement, but we still are inclining to introducing the language clause.

Reply:
Pretty much all other databases providing SQL function support do not require this, which is partly why I think it's probably not necessary. At this point I think it's ok if you want to introduce this but make it optional, same as `LANGUAGE SQL` is optional. Though SQL spec says it should be specified, I don't see a point. And unlike other SQL statement, `CREATE FUNCTION` doesn't have a strict backward compatibility issue if you ever need to require this to be specified.